crates/rover-client/src/operations/persisted_queries/publish/types.rs

```diff
@@ -152,3 +234,139 @@ impl PersistedQueriesOperationCounts {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn it_can_read_relay_manifest() {
+        let relay_manifest = r#"{
+      "ed145403db84d192c3f2f44eaa9bc6f9": "query NewsfeedQuery {\n  topStory {\n    title\n    summary\n    poster {\n      __typename\n      name\n      profilePicture {\n        url\n      }\n      id\n    }\n    thumbnail {\n      url\n    }\n    id\n  }\n}\n"
+    }"#;
+
+        let relay_manifest: RelayPersistedQueryManifest =
+            serde_json::from_str(relay_manifest).expect("could not read relay manifest");
+        assert_eq!(relay_manifest.operations.len(), 1);
+    }
+
+    #[test]
+    fn it_can_convert_relay_manifest_to_apollo_manifest() {
+        let id = "ed145403db84d192c3f2f44eaa9bc6f9";
+        let body = "query NewsfeedQuery {\n  topStory {\n    title\n    summary\n    poster {\n      __typename\n      name\n      profilePicture {\n        url\n      }\n      id\n    }\n    thumbnail {\n      url\n    }\n    id\n  }\n}\n";
+        let relay_manifest = serde_json::json!({id: body}).to_string();
+
+        let relay_manifest: RelayPersistedQueryManifest =
+            serde_json::from_str(&relay_manifest).expect("could not read relay manifest");
+        let apollo_manifest: ApolloPersistedQueryManifest = relay_manifest.try_into().unwrap();
+        assert_eq!(
+            apollo_manifest.operations[0],
+            PersistedQueryOperation {
+                name: "NewsfeedQuery".to_string(),
+                r#type: PersistedQueryOperationType::Query,
+                id: id.to_string(),
+                body: body.to_string()
+            }
+        );
+    }
+
+    #[test]
+    fn it_can_convert_relay_manifest_with_multiple_queries_to_apollo_manifest() {
+        let id_one = "ed145403db84d192c3f2f44eaa9bc6f9";
+        let body_one = "query NewsfeedQuery {\n  topStory {\n    title\n    summary\n    poster {\n      __typename\n      name\n      profilePicture {\n        url\n      }\n      id\n    }\n    thumbnail {\n      url\n    }\n    id\n  }\n}\n";
+        let id_two = "adkjflaskdjf";
+        let body_two = "mutation NamedMutation { topStory }";
+        let relay_manifest = serde_json::json!({id_one: body_one, id_two: body_two}).to_string();
+
+        let relay_manifest: RelayPersistedQueryManifest =
+            serde_json::from_str(&relay_manifest).expect("could not read relay manifest");
+        let apollo_manifest: ApolloPersistedQueryManifest = relay_manifest.try_into().unwrap();
+        assert_eq!(
```

Comment:
This test is failing; it looks like there's an assumption of what order the operations will end up in the manifest that isn't valid.

Reply:
fixed